compose/smash/Dockerfile

```diff
@@ -1,19 +1,29 @@
-ARG UBUNTU_VERSION=20.04
+FROM nixos/nix AS nix-configured
```

Comment:
If you want a minimal image (say, based on Alpine), you could build a _static_ binary of the smash server, and copy only it (no dependencies needed) from the build stage to a runtime stage. This would only replace the part downloading the binary.

If needed, I can find a Nix attribute to build the static self-contained binary

Reply:
@michalrus That's a great idea! I couldn't find how to 'package' a stand-alone binary. Could you help me with it?

Reply:
Yes, I don't know off the top of my head, but let's try :thinking:

Reply:
I think the attribute is `static/cardano-smash-server`, so it will be something like:

```
RUN nix build -L github:input-output-hk/cardano-db-sync/${CARDANO_DB_SYNC_VERSION:-13.1.0.0}#static/cardano-smash-server

FROM alpine:… as runtime

COPY --from=buildtime /where/you/built/result/cardano-smash-server /bin/cardano-smash-server

RUN cardano-smash-server --version
```

Reply:
Hmm, that won't work before because there are linked libraries as well. (we are compiling some libraries from the source while )

Reply:
Yeah, you'd need to use this statically-linked binary instead: `#static/cardano-smash-server` – it is self-contained and has all the (previously shared) libraries inside

Reply:
@michalrus done in a908181e4f6b6d438a66cb2f8da5844425f3f8dc